tests/test_server_proto.py

```diff
@@ -2827,6 +2828,63 @@ async def test_server_proto_query_cache_invalidate_09(self):
         finally:
             await self.con.query('ROLLBACK')
 
+    async def test_server_proto_query_cache_invalidate_10(self):
+        typename = 'CacheInv_10'
+
+        con1 = self.con
+        con2 = await self.connect(database=con1.dbname)
+        try:
+            await con2.execute(f'''
+                CREATE TYPE {typename} {{
+                    CREATE REQUIRED PROPERTY prop1 -> std::str;
+                }};
+
+                INSERT {typename} {{
+                    prop1 := 'aaa'
+                }};
+            ''')
+
+            sleep = 3
+            query = (
+                f"# EDGEDB_TEST_COMPILER_SLEEP = {sleep}\n"
+                f"SELECT {typename}.prop1"
+            )
+            task = self.loop.create_task(con1.query(query))
+
+            start = time.monotonic()
+            await con2.execute(f'''
+                DELETE (SELECT {typename});
+
+                ALTER TYPE {typename} {{
+                    DROP PROPERTY prop1;
+                }};
+
+                ALTER TYPE {typename} {{
+                    CREATE REQUIRED PROPERTY prop1 -> std::int64;
+                }};
+
+                INSERT {typename} {{
+                    prop1 := 123
+                }};
+            ''')
+            if time.monotonic() - start > sleep:
+                self.skipTest("The host is too slow for this test.")
+                # If this happens too much, consider increasing the sleep time
+
+            with self.assertRaisesRegex(
+                edgedb.errors.InternalServerError,
+                "column .* does not exist",
+            ):
```

Comment:
This seems wrong as intended behavior. At least in this case, where the schema is changed while we were compiling, it should be easy to notice that and retry.

It does seem like it might be hard to prevent in the general case, though, without locking things in ways we don't want to? Probably worth thinking about.

Reply:
Created an issue for a future fix: https://github.com/edgedb/edgedb/issues/6820